src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1333,9 +1392,40 @@ public void InvalidateHttp2Connection(Http2Connection connection)
         {
             lock (SyncObj)
             {
-                if (_http2Connection == connection)
+                Http2Connection[]? localHttp2Connections = _http2Connections;
```

Comment:
nit: Do you need localHttp2Connections variable while holding SyncObj? You shouldn't need to worry about someone else modifying the array.

Reply:
`_http2Connections` is now marked as `volatile` because it's read outside locks, so having a local copy avoids multiple reads of volatile field to improve performance.